File: Gost/c++/generator.cpp
#include"header.h"
#include<time.h>
#include<stdlib.h>

using namespace std;


void swap(int*array,int pos1,int pos2){
    int temp = array[pos1];
    array[pos1] = array[pos2];
    array[pos2] = temp;
}

void shuffle(int*data){
    int* array = new int[16];
    for (int i = 0; i < 16; ++i) {
        array[i] =  rand();
    }
    for (int i = 0; i < 16; ++i)
        for (int j = 0; j < 16 - i -1; ++j)
           if(array[j] > array[j+1]){
               swap(array,j,j+1);
               swap(data,j,j+1);
           }
}


void generate_matrix(ofstream *file) {
    int* row = new int[16]{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15};
    for (int i = 0; i < 8; ++i) {
        shuffle(row);
        for(int i =0;i<16;++i)
            cout<<row[i]<<" ";
        cout << endl;
        for (int j = 0; j < 16; ++j) {
            int m =  (row[j] << 4 * i);
            file->write(reinterpret_cast<const char *>(&m), 4);
        }
    }

}

void generate_keys(ofstream *file){
    for(int i = 0;i < 8 ; ++i){
        int r = rand();
        file->write(reinterpret_cast<const char *>(&r), 4);
    }
}


int  generate(){
    srand(time(nullptr));
    ofstream out("./info",ios_base::binary);
    generate_keys(&out);
    generate_matrix(&out);
    out.close();
    return 0;
}

File: Gost/java/src/Generator.java
import nokinobi.stream.LittleEndianOutputStream;

import java.io.DataOutputStream;
import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.nio.ByteBuffer;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.Random;

public class Generator {

    public static void main(String[] args) throws IOException {
        int[][] matrix = generate_matrix();
        int[] keys = generate_keys();
        File file = new File("./info");
        LittleEndianOutputStream out = new LittleEndianOutputStream(new FileOutputStream(file));
        for (int key : keys)
            out.write(key);
        for (int i =0; i < 8; ++i)
            for (int j = 0; j < 16; j++)
                out.write(matrix[j][i]);
        out.close();
    }

    private static int[] generate_keys(){
        Random rand = new Random();
        int[] keys = new int[8];
        for (int i =0 ; i < 8 ; i++)
            keys[i] = rand.nextInt();
        return  keys;
    }

    private static int[][] generate_matrix(){
        int [][] matrix = new int[16][8];
        for(int i = 0 ; i < 8 ; i++){
            List<Integer> row = Arrays.asList(0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15);
            Collections.shuffle(row);
            for(int j =0 ; j < row.size();j++)
                matrix[j][i] = row.get(j) << i*4;
        }
        return matrix;
    }
}
